conbench/app/benchmarks.py

```diff
@@ -180,17 +194,38 @@ def show_trends_for_benchmark(bname: TBenchmarkName) -> str:
         # datetimeindex back to float values. Remove 10^15 from these
         # nanoseconds-since-epoch to make the numeric values a little less
         # extreme for printing/debugging.
+
+        remove_outliers_by_iqrdist(df, "svs")
+
+        # Now it's important to drop nans again because the outliers have been
+        # marked with NaN, and any NaN will nannify the linear fit. We do not
+        # want to mutate the DF in the BMRT cache. drop nans, create explicit
+        # copy (otherwise this might be a view)
+        df = df.dropna().copy()  # drop all rows that have any NaN
+        if len(df.index) < 10:
+            # Skip if after outlier removal there's not enough history left.
+            continue
+
         tfloats = (
             np.array(df.index.to_pydatetime(), dtype=np.datetime64).astype("float")
             / 10**15
         )
         yfloats = df["svs"].values
 
+        # Note that this is a least squares fit.
```

Comment:
also note: ideally this for loop is in the future a single matrix operation. it doesn't quite make sense to iterate over many timeseries individually and to do the same analysis logic one-by-one within a Python loop. That's cries for optimization via proper usage of linear algebra / pandas dataframe operation, will probably speed this up by several orders of magnitude (and this might be needed, this takes already a second for 1000 timeseries).